salt/proxy/junos.py

```diff
@@ -118,10 +120,18 @@ def conn():
 
 def alive(opts):
     '''
-    Return the connection status with the remote device.
+    Validate and return the connection status with the remote device.
 
     .. versionadded:: Oxygen
     '''
+    try:
+        thisproxy['conn'].rpc('<file-list><path>/dev/null</path></file-list>')
```

Comment:
Suggest changing this line to something like:
```
thisproxy['conn'].rpc.file_list(path='/dev/null', dev_timeout=some_value)
```

I would need a better understanding to make a recommendation on what `some_value` should be, but I suspect you want something less than the default of 30 seconds.